src/main/java/data_gateway/notification/JsonNotificationAdapter.java

```diff
@@ -24,7 +26,30 @@ public void write(JsonWriter jsonWriter, Notification notification) throws IOExc
 
 
     @Override
-    public Notification read(JsonReader jsonReader) {
+    public Notification read(JsonReader jsonReader) throws IOException {
+        long id = 0;
```

Comment:
I'm sure that we'll never actually get a 0 id with the Snowflake, but because 0 is actually a valid id it would be nice to have either something invalid (maybe negative?) or null. That way this seems less like a  "default value" and more of a "special invalid value."
It also helps in handling invalid cases if that ever needs to be a thing (but for the most part it's implicit that the initial values will be replaced).